### PR TITLE
[tensorflow] Fix a crash caused by tf::skipInternalLocations.

### PR DESCRIPTION
**Pull request by dingobye**
This patch improves `tf::skipInternalLocations` to correctly handle non-inlined SILDebugScope input. It returns the input `loc` in this case, instead of a new `SILDebugLocation(ds->Loc, ds)` which may lead to crashes in SIL verification.

Resolves [SR-8419](https://bugs.swift.org/browse/SR-8419).

### CODE REVIEW

**Comment by dingobye** [comment]
@dan-zheng Hi Dan, would you take a look, please?

File: test/TensorFlow/crashers.swift [hunk 1]
```diff
@@ -353,3 +353,17 @@ public func constFoldingBug() {
     return
   }
 }
+
+public func SR8419(iterationCount: Int) {
```
Comment (dan-zheng) [inline test/TensorFlow/crashers.swift]: Could you please add a comment before this function explaining the initial crash?
Sth like: `// This function failed SIL verification due to a non-inlined SILDebugLocation.`
Thanks!

**Review by dan-zheng — APPROVED** [review]
LGTM, thank you! 😄

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Review by rxwei — APPROVED** [review]
LGTM!

File: test/TensorFlow/crashers.swift (hunk 1 above)
Comment (dingobye) [reply test/TensorFlow/crashers.swift]: Fixed

**Comment by dingobye** [comment]
Thank you for your review, Dan and Richard! I update the code as requested.

**Comment by rxwei** [comment]
@swift-ci Please test tensorflow

**Comment by rxwei** [comment]
@swift-ci Please test tensorflow

**Comment by rxwei** [comment]
@swift-ci Please test tensorflow

**Comment by rxwei** [comment]
@swift-ci Please test tensorflow

**Comment by rxwei** [comment]
Thanks @dingobye !

**Pull request merged**